examples/print-dialog/print_dialog.py

```diff
@@ -0,0 +1,540 @@
+"""
+Demo application of a Print Upload Dialog using the Formlabs Local API.
+
+The application is meant to be demo of the API calls and logic involved in building a print upload dialog.
+The model to be sliced and uploaded is a simple cube.stl file included in the same directory as this script.
+
+The following are all left intentionally unfinished to keep the demo code concise:
+- Loading user models
+- Print preparation
+- Application styling
+- Deployment
+
+Installation:
+- On MacOS if using Python 3.12: brew install python-tk@3.12
+"""
+
+import tkinter as tk
+from tkinter import messagebox, simpledialog
+from tkinter import ttk  # themed tk
+from typing import TypedDict, NotRequired
+import atexit
+import pathlib
+from requests import request
+from requests.exceptions import HTTPError
+import formlabs_local_api_minimal as formlabs
+import sys
+import time
+
+server_path: str = ""
+if sys.platform == "win32":
+    server_path = str(pathlib.Path().resolve().parents[1] / "PreFormServer/PreFormServer.exe")
+elif sys.platform == "darwin":
+    server_path = str(pathlib.Path().resolve().parents[1] / "PreFormServer.app/Contents/MacOS/PreFormServer")
+else:
+    print("Unsupported platform")
+    sys.exit(1)
+
+# Type Definitions
+PrinterTypeDropdownLabel = str  # Example "Form 3/3+"
+PrinterDropdownLabel = str  # Example [Form 3] TestyTest - Idle
+PrintSettingDropdownLabel = str  # Example "Alumina 4N V1 0.050 mm (Default Settings)"
+ProductName = str  # Example: "Form 3"
+MachineTypeId = str  # Example: "FORM-3-0"
+PrintOperationId = str  # uuid string
+DeviceData = TypedDict(
+    "DeviceData",
+    {
+        "product_name": ProductName,
+        "id": str,
+        "status": str,
+        "dashboard_queue_id": NotRequired[str],
+        "supported_machine_type_ids": NotRequired[list[MachineTypeId]],
+        "ready_to_print_now": NotRequired[bool],
+    },
+)
+SceneSettings = TypedDict(
+    "SceneSettings",
+    {
+        "layer_thickness_mm": float,
+        "machine_type": MachineTypeId,
+        "material_code": str,
+        "print_setting": str,
+    },
+)
+
+TEST_FILE_PATH = pathlib.Path().resolve() / "cube.stl"
```

Comment:
```suggestion
parser = argparse.ArgumentParser(description="open print dialog for a specific .stl file")
parser.add_argument("model", type=str, help="Path to the STL", nargs='?')
args = parser.parse_args()

TEST_FILE_PATH = pathlib.Path().resolve() / "cube.stl"
modelFilePath = args.model if args.model else TEST_FILE_PATH
```

Maybe cube.stl could live in formlabs-api-python/examples/sample-models or somesuch?

Reply:
I'd like to keep the implementation as-is if you don't mind because configurability is not a goal of the code and I think referencing a local file is the simplest solution.